Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,179 @@ protected override void ProcessRecord()
+        /// <para type="description">
+        /// The lifetime of this table from the time of creation (in seconds).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValueWithRef)]
+        [ValidateRange(1, (long.MaxValue / 1000))]
+        public long Expiration { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            TablesResource.InsertRequest request;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    request = Service.Tables.Insert(InputObject,
+                        InputObject.TableReference.ProjectId,
+                        InputObject.TableReference.DatasetId);
+                    break;
+                case ParameterSetNames.ByValueWithRef:
+                    Project = Dataset.ProjectId;
+                    DatasetId = Dataset.DatasetId;
+                    request = makeInsertReq();
+                    break;
+                case ParameterSetNames.ByValues:
+                    request = makeInsertReq();
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            // Add the new dataset to the project supplied.
+            try
+            {
+                Table response = request.Execute();
+                WriteObject(response);
+            }
+            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Conflict)
+            {
+                WriteError(new ErrorRecord(ex,
+                    $"A table with the name {TableId} already exists in {Project}:{DatasetId}.",
+                    ErrorCategory.InvalidArgument,
+                    TableId));
+            }
+        }
+
+        public TablesResource.InsertRequest makeInsertReq()
+        {
+            Table newTab;
+            newTab = new Table();
+            newTab.TableReference = getTableRef(Project, DatasetId, TableId);
+            newTab.FriendlyName = Name;
+            newTab.Description = Description;
+            if (Expiration != 0)
+            {
+                long currentMillis = Convert.ToInt64(DateTime.Now.ToUniversalTime().Subtract(
+                    new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)
+                    ).TotalMilliseconds);
+                newTab.ExpirationTime = (Expiration * 1000) + currentMillis;
+                // Note: The code below is a more elegant solution, but is not supported by the current version
+                //newTab.ExpirationTime = new DateTimeOffset(DateTime.Now).ToUnixTimeMilliseconds();
+            }
+            return Service.Tables.Insert(newTab, Project, DatasetId);
+        }
+
+        public TableReference getTableRef(string p, string d, string t)
```

Comment:
These names should be full words: projectId, datasetId, tableId. The local variable should also probably be tableReference, but the function is short enough that it is not as important as the parameter names.